test/metabase/api/preview_embed_test.clj

```diff
@@ -141,6 +141,20 @@
              (mt/user-http-request :crowberto :get 202 (str (card-query-url card {:_embedding_params {:venue_id "enabled"}})
                                                             "?venue_id=200")))))))))
 
+(deftest query-max-results-constraint-test
+  (testing "GET /api/preview_embed/card/:token/query"
+    (testing "Only 2000 results returned when there are many more"
+      (embed-test/with-embedding-enabled-and-new-secret-key
+        (let [;; this is the same sample question as setup in these steps to reproduce:
+              ;; https://github.com/metabase/metabase/issues/20938
+              sample-db-orders-question {:database 1
+                                         :query {:source-table 2}
+                                         :type "query"}]
+          (embed-test/with-temp-card [card {:dataset_query sample-db-orders-question}]
+            (is (>= 2000 (-> (mt/user-http-request :crowberto :get 202 (card-query-url card))
```

Comment:
I think this would be a bit more readable if you wrote it at `(<= row-count 2000)`. Otherwise it's too much of a [Yoda condition](https://en.wikipedia.org/wiki/Yoda_conditions). You would read `(>= 2000 x)` as `2000 >= x` which is a little _weird_.  `x <= 2000` is more natural to read. I know equality tests we do `(is (= expected actual))` but this is a normal predicate test expression so I think the Yoda condition rule applies here.

I'm only brining this up because I did a double take and spent a couple minutes reading this to make sure it was right when I saw it. YMMV

Reply:
Yeah, as you could tell I was matching the `(is (= expected actual))` pattern. This has been changed so as to not use `<=` or `>=`.